### PR TITLE
process: use PT_GNU_STACK elf section size on MMU targets

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

Solving issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/496

RTOS-304

## Motivation and Context

Custom stack can be used on MMU targets:
`LOCAL_LDFLAGS := -z stack-size=0x2000 -z noexecstack`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (zynq7000-qemu, riscv64-qemu, ia32-qemu).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: proc/process.c [hunk 1]
```diff
@@ -492,28 +505,32 @@ int process_load(process_t *process, vm_object_t *o, offs_t base, size_t size, v
...
+		default:
+			err = -ENOEXEC;
 	}
 	vm_munmap(process_common.kmap, ehdr, size);
 
 	if (err < 0)
 		return err;
 
+	lib_printf("Stack size 0x%x\n", ustacksz);
+
```
Comment (agkaminski) [inline proc/process.c]:
```suggestion
```

File: proc/process.c [hunk 2]
```diff
@@ -474,6 +486,7 @@ int process_load(process_t *process, vm_object_t *o, offs_t base, size_t size, v
 	void *stack;
 	Elf64_Ehdr *ehdr;
 	vm_map_t *map = process->mapp;
+	size_t ustacksz;
```
Comment (agkaminski) [inline proc/process.c:489]: I guess it's a matter of taste where default value is assigned. As an optional change I like the idea of subfunction optionally changing it's value. Just a suggestion
```suggestion
	size_t ustacksz = SIZE_USTACK;
```

**Review by agkaminski — APPROVED** [review]

**Comment by nalajcie** [comment]
side note: if You would use `Fixes: [gh-issue]` or `Solves: [gh-issue]` then the linked issue would automatically be resolved on PR merge (see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Comment by HBuczynski** [comment]
@nalajcie, I know it but I have forgotten to use it ;) 
hahaha, I was sure that you will spot it and give a note ;-)

**Pull request merged**